tests/union_vector/Movie.java

```diff
@@ -62,5 +62,105 @@ public static int endMovie(FlatBufferBuilder builder) {
     public Movie get(int j) { return get(new Movie(), j); }
     public Movie get(Movie obj, int j) {  return obj.__assign(__indirect(__element(j), bb), bb); }
   }
+  public MovieT unpack() {
+    MovieT _o = new MovieT();
+    unpackTo(_o);
+    return _o;
+  }
+  public void unpackTo(MovieT _o) {
+    CharacterUnion _oMainCharacter = new CharacterUnion();
+    byte _oMainCharacterType = mainCharacterType();
+    _oMainCharacter.setType(_oMainCharacterType);
+    Table _oMainCharacterValue;
+    switch (_oMainCharacterType) {
+      case Character.MuLan:
+        _oMainCharacterValue = mainCharacter(new Attacker());
+        _oMainCharacter.setValue(_oMainCharacterValue != null ? ((Attacker) _oMainCharacterValue).unpack() : null);
+        break;
+      case Character.Rapunzel:
+        _oMainCharacterValue = mainCharacter(new Rapunzel());
+        _oMainCharacter.setValue(_oMainCharacterValue != null ? ((Rapunzel) _oMainCharacterValue).unpack() : null);
+        break;
+      case Character.Belle:
+        _oMainCharacterValue = mainCharacter(new BookReader());
+        _oMainCharacter.setValue(_oMainCharacterValue != null ? ((BookReader) _oMainCharacterValue).unpack() : null);
+        break;
+      case Character.BookFan:
+        _oMainCharacterValue = mainCharacter(new BookReader());
+        _oMainCharacter.setValue(_oMainCharacterValue != null ? ((BookReader) _oMainCharacterValue).unpack() : null);
+        break;
+      case Character.Other:
+        _oMainCharacterValue = mainCharacter(new String());
+        _oMainCharacter.setValue(_oMainCharacterValue != null ? ((String) _oMainCharacterValue).unpack() : null);
+        break;
+      case Character.Unused:
+        _oMainCharacterValue = mainCharacter(new String());
+        _oMainCharacter.setValue(_oMainCharacterValue != null ? ((String) _oMainCharacterValue).unpack() : null);
```

Comment:
@aardappel @paulovap This generated code fails due to incorrect implementation of Union in Java.
I described it in issue #6561
I suggest to remove the tests in JavaTest.java and the matching shell and batch scripts related to union_vector until this issue is resolved.

Reply:
@ofirm93 we need to have all the tests good to go to merge this PR. Its not good for us to merge it without proper testing.

I will try to find some time on the weekend to work on #6561. Once we have that merged we can merge your PR as well with the tests.

Reply:
@paulovap Can we deliver assuming ObjectAPI is in alpha stage or doesn't support Unions?
Because the blocker is unrelated to the PR itself.
I think that the solution for the Unions problem is much more complex, than we 
think and should handle backword compatibility which can cause real delay for this PR.
I will continue support for this feature and I wish to try it in real life projects.

Reply:
@ofirm93, can you add unit tests for everything else other than the union issue? Than later we can look into the union problem. But adding this feature without any kind of unit test is not a good approach.

Reply:
@paulovap I already did it. Look at the JavaTest.java file for the changes

Reply:
> I suggest to remove the tests in JavaTest.java and the matching shell and batch scripts related to union_vector until this issue is resolved.

Ah alright, I misunderstood that.